dist/tools/jlink/jlink.sh

```diff
@@ -58,7 +58,12 @@ _JLINK_SERVER=JLinkGDBServer
 _JLINK_IF=SWD
 _JLINK_SPEED=2000
 # default starting address of the devices flash memory
-_FLASH_ADDR=0
+if [ -n "${FLASH_OFFSET}" ]; then
+    _FLASH_ADDR=${FLASH_OFFSET}
+    echo "Flashing at offset ${FLASH_OFFSET}"
+else
+    _FLASH_ADDR=0
+fi
```

Comment:
this seems unclean, you override the default flash address variable just for it to be checked again below... So now we use 2 different 'external' variables (`FLASH_OFFSET` and `JLINK_FLASH_ADDR`) do determine the actual used flash address... Even worse, if `JLINK_FLASH_ADDR` is defined by the CPU, then the `FLASH_OFFSET` variable will be ignored!

So this needs to be cleaned up.